dashboard/app/models/user_level.rb

```diff
@@ -25,19 +25,29 @@
 # Summary information about a User's Activity on a Level in a Script.
 # Includes number of attempts (attempts), best score and whether it was submitted
 class UserLevel < ActiveRecord::Base
+  AUTOLOCK_PERIOD = 1.day
+
   belongs_to :user
   belongs_to :level
   belongs_to :script
   belongs_to :level_source
 
   before_save :handle_unsubmit
 
+  validate :readonly_requires_submitted
+
   # TODO(asher): Consider making these scopes and the methods below more consistent, in tense and in
   # word choice.
   scope :attempted, -> { where.not(best_result: nil) }
   scope :passing, -> { where('best_result >= ?', ActivityConstants::MINIMUM_PASS_RESULT) }
   scope :perfect, -> { where('best_result > ?', ActivityConstants::MAXIMUM_NONOPTIMAL_RESULT) }
 
+  def readonly_requires_submitted
+    if readonly_answers? && !submitted?
+      errors.add(:readonly_answers, 'readonly_answers only valid on submitted UserLevel')
```

Comment:
How will this error be surfaced?  My concern is that most places we update `UserLevel` are in async actions, or somewhere the user (or us) won't see the error.